## 1. Importing libraries

In [1]:
!pip install numpy
!pip install pandas 
!pip install scikitlearn

Defaulting to user installation because normal site-packages is not writeable



[notice] A new release of pip is available: 24.0 -> 25.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip


Defaulting to user installation because normal site-packages is not writeable



[notice] A new release of pip is available: 24.0 -> 25.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip


Defaulting to user installation because normal site-packages is not writeable


ERROR: Could not find a version that satisfies the requirement scikitlearn (from versions: none)
ERROR: No matching distribution found for scikitlearn

[notice] A new release of pip is available: 24.0 -> 25.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
!pip install scikit-learn

Defaulting to user installation because normal site-packages is not writeable



[notice] A new release of pip is available: 24.0 -> 25.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
import os
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split

## 2. Reading the data

In [4]:
PROJECT_DIR = r"C:\Users\Dell\OneDrive\Desktop\flight-sagemaker-project" # the r over here stands for 'raw'
DATA_DIR = "data"

In [5]:
# os.path.join(PROJECT_DIR, DATA_DIR, "file_price.csv")

In [6]:
def get_data(name):
    file_name = f"{name}.csv"
    file_path = os.path.join(PROJECT_DIR, DATA_DIR, file_name)
    return pd.read_csv(file_path)

In [7]:
flights = get_data("flight_price")
flights.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

In [8]:
flights.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10683 entries, 0 to 10682
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Airline          10683 non-null  object
 1   Date_of_Journey  10683 non-null  object
 2   Source           10683 non-null  object
 3   Destination      10683 non-null  object
 4   Route            10682 non-null  object
 5   Dep_Time         10683 non-null  object
 6   Arrival_Time     10683 non-null  object
 7   Duration         10683 non-null  object
 8   Total_Stops      10682 non-null  object
 9   Additional_Info  10683 non-null  object
 10  Price            10683 non-null  int64 
dtypes: int64(1), object(10)
memory usage: 918.2+ KB


In [9]:
flights.isnull().sum()

Airline            0
Date_of_Journey    0
Source             0
Destination        0
Route              1
Dep_Time           0
Arrival_Time       0
Duration           0
Total_Stops        1
Additional_Info    0
Price              0
dtype: int64

- The dataset contains 10,683 rows  and 11 features 
- Columns 'Route' and 'Total_Stops' have missing value each 
- Datatypes of some features isn't appropriate

# 3. Preliminary analysis

### 3.1 Check Data Types 

In [10]:
flights.head()

Airline Date_of_Journey    Source Destination                  Route  \
0       IndiGo      24/03/2019  Banglore   New Delhi              BLR → DEL   
1    Air India       1/05/2019   Kolkata    Banglore  CCU → IXR → BBI → BLR   
2  Jet Airways       9/06/2019     Delhi      Cochin  DEL → LKO → BOM → COK   
3       IndiGo      12/05/2019   Kolkata    Banglore        CCU → NAG → BLR   
4       IndiGo      01/03/2019  Banglore   New Delhi        BLR → NAG → DEL   

  Dep_Time  Arrival_Time Duration Total_Stops Additional_Info  Price  
0    22:20  01:10 22 Mar   2h 50m    non-stop         No info   3897  
1    05:50         13:15   7h 25m     2 stops         No info   7662  
2    09:25  04:25 10 Jun      19h     2 stops         No info  13882  
3    18:05         23:30   5h 25m      1 stop         No info   6218  
4    16:50         21:35   4h 45m      1 stop         No info  13302

To check the datatypes 👇

In [11]:
flights.dtypes

Airline            object
Date_of_Journey    object
Source             object
Destination        object
Route              object
Dep_Time           object
Arrival_Time       object
Duration           object
Total_Stops        object
Additional_Info    object
Price               int64
dtype: object

In [12]:
value = flights.Date_of_Journey.loc[6]
value

'12/03/2019'

❌ the feature "Date_of_Journey" contains values in string format, which is incorrect. 

In [13]:
value = flights.Dep_Time.iloc[6]
value

'18:55'

### 3.2 Check for duplicates 

In [14]:
flights.duplicated().sum()

220

☝️The above is the result of total no. duplicate rows in the dataset.

In [15]:
(
    flights
    .loc[flights.duplicated(keep=False)]
    .sort_values(['Airline','Date_of_Journey','Source','Destination'])
    
)

Airline Date_of_Journey    Source Destination                  Route  \
6321  Air India      01/03/2019  Banglore   New Delhi  BLR → BOM → AMD → DEL   
9848  Air India      01/03/2019  Banglore   New Delhi  BLR → BOM → AMD → DEL   
572   Air India      03/03/2019  Banglore   New Delhi              BLR → DEL   
8168  Air India      03/03/2019  Banglore   New Delhi              BLR → DEL   
1495  Air India       1/04/2019   Kolkata    Banglore  CCU → DEL → COK → BLR   
...         ...             ...       ...         ...                    ...   
2692   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
2870   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
3711   SpiceJet      24/03/2019  Banglore   New Delhi              BLR → DEL   
2634    Vistara      24/03/2019  Banglore   New Delhi              BLR → DEL   
5470    Vistara      24/03/2019  Banglore   New Delhi              BLR → DEL   

     Dep_Time  Arrival_Time Duration Total_Stops  \
6321    08:50  23:55 02 Mar   39h 5m     2 stops   
9848    08:50  23:55 02 Mar   39h 5m     2 stops   
572     21:10         23:55   2h 45m    non-stop   
8168    21:10         23:55   2h 45m    non-stop   
1495    10:00  01:20 02 Apr  15h 20m     2 stops   
...       ...           ...      ...         ...   
2692    05:45         08:35   2h 50m    non-stop   
2870    05:45         08:35   2h 50m    non-stop   
3711    20:30         23:20   2h 50m    non-stop   
2634    11:30         14:10   2h 40m    non-stop   
5470    11:30         14:10   2h 40m    non-stop   

                   Additional_Info  Price  
6321                       No info  17135  
9848                       No info  17135  
572                        No info   7591  
8168                       No info   7591  
1495                       No info  10408  
...                            ...    ...  
2692  No check-in baggage included   4273  
2870  No check-in baggage included   4273  
3711  No check-in baggage included   3873  
2634                       No info   5403  
5470                       No info   5403  

[415 rows x 11 columns]

### 3.3 Make observations

- the datatype of 'Date_of_Journey', 'Dep_Time' should be changed to datetime
- the type of 'Duration' and 'Total_Stops' is mixed. It should be numeric .
- There are 220 duplicate values . These should be removed

# 4. Detailed Analysis 

Airline

In [16]:
flights.Airline.unique()

array(['IndiGo', 'Air India', 'Jet Airways', 'SpiceJet',
       'Multiple carriers', 'GoAir', 'Vistara', 'Air Asia',
       'Vistara Premium economy', 'Jet Airways Business',
       'Multiple carriers Premium economy', 'Trujet'], dtype=object)

- Some of the entries have inconsistent/inaccurate values

In [55]:
flights.drop(index=flights.index[[550]],inplace=True)

☝️ here we are dropping the row containing the info for Trujet airline because there in only 1 entry for this airline 

In [56]:
(
    flights
    .Airline
    .str.replace(" Premium economy", "")
    # ☝️ here we replaced the string 'Premium economy' with empty string
    .str.replace(" Business", "")
    # ☝️ here we replaced the string 'Business' with empty string
    .str.title()
    # ☝️ the first letter of the titles will be made capital . i.e carriers->Carriers
    #.drop(index=flights.index[[550]],inplace=True)
    .unique()
)

array(['Indigo', 'Air India', 'Jet Airways', 'Spicejet',
       'Multiple Carriers', 'Goair', 'Vistara', 'Air Asia', 'Trujet'],
      dtype=object)

## Date_Of_Journey

In [57]:
flights.Date_of_Journey

0        24/03/2019
1         1/05/2019
2         9/06/2019
3        12/05/2019
4        01/03/2019
            ...    
10678     9/04/2019
10679    27/04/2019
10680    27/04/2019
10681    01/03/2019
10682     9/05/2019
Name: Date_of_Journey, Length: 10682, dtype: object

In [58]:
pd.to_datetime(flights.Date_of_Journey, dayfirst=True)
# The dayfirst feature is to handle or check if the argument is string or not 

0       2019-03-24
1       2019-05-01
2       2019-06-09
3       2019-05-12
4       2019-03-01
           ...    
10678   2019-04-09
10679   2019-04-27
10680   2019-04-27
10681   2019-03-01
10682   2019-05-09
Name: Date_of_Journey, Length: 10682, dtype: datetime64[ns]

## Source

In [59]:
flights.Source.unique()

array(['Banglore', 'Kolkata', 'Delhi', 'Chennai', 'Mumbai'], dtype=object)

## Destination

In [60]:
flights.Destination.unique()

array(['New Delhi', 'Banglore', 'Cochin', 'Kolkata', 'Delhi', 'Hyderabad'],
      dtype=object)

## Dep_time

In [61]:
flights.Dep_Time

0        22:20
1        05:50
2        09:25
3        18:05
4        16:50
         ...  
10678    19:55
10679    20:45
10680    08:20
10681    11:30
10682    10:55
Name: Dep_Time, Length: 10682, dtype: object

In [62]:
(
    flights
    .Dep_Time
    .loc[lambda ser: ser.str.contains("[^0-9:]")]
    #          ☝️                      ☝️
    #     flights.Dep_Time   we want all the rows apart from this 
)

Series([], Name: Dep_Time, dtype: object)

In [63]:
pd.to_datetime(flights.Dep_Time)

C:\Users\Dell\AppData\Local\Temp\ipykernel_1260\2682404337.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  pd.to_datetime(flights.Dep_Time)


0       2025-05-30 22:20:00
1       2025-05-30 05:50:00
2       2025-05-30 09:25:00
3       2025-05-30 18:05:00
4       2025-05-30 16:50:00
                ...        
10678   2025-05-30 19:55:00
10679   2025-05-30 20:45:00
10680   2025-05-30 08:20:00
10681   2025-05-30 11:30:00
10682   2025-05-30 10:55:00
Name: Dep_Time, Length: 10682, dtype: datetime64[ns]

In [64]:
pd.to_datetime(flights.Dep_Time).dt.time

C:\Users\Dell\AppData\Local\Temp\ipykernel_1260\1056054517.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  pd.to_datetime(flights.Dep_Time).dt.time


0        22:20:00
1        05:50:00
2        09:25:00
3        18:05:00
4        16:50:00
           ...   
10678    19:55:00
10679    20:45:00
10680    08:20:00
10681    11:30:00
10682    10:55:00
Name: Dep_Time, Length: 10682, dtype: object

## Arival_Time

In [65]:
flights.Arrival_Time

0        01:10 22 Mar
1               13:15
2        04:25 10 Jun
3               23:30
4               21:35
             ...     
10678           22:25
10679           23:20
10680           11:20
10681           14:10
10682           19:15
Name: Arrival_Time, Length: 10682, dtype: object

In [66]:
(
    flights
    .Arrival_Time
    .loc[ lambda ser: ser.str.contains("[^0-9:]")]
    .str.split(" ",n=1)
    .str.get(1)
    .unique()
)

array(['22 Mar', '10 Jun', '13 Mar', '02 Mar', '10 May', '04 Mar',
       '13 Jun', '28 May', '19 Mar', '07 May', '02 Jun', '16 Jun',
       '19 May', '16 May', '28 Jun', '02 May', '28 Mar', '19 Jun',
       '04 Apr', '25 Mar', '07 Mar', '25 Jun', '07 Jun', '25 May',
       '13 May', '16 Mar', '22 May', '10 Apr', '04 Jun', '20 May',
       '28 Apr', '25 Apr', '10 Mar', '19 Apr', '13 Apr', '02 Apr',
       '23 Mar', '22 Apr', '11 May', '07 Apr', '03 May', '08 Mar',
       '03 Mar', '05 Mar', '22 Jun', '04 May', '26 May', '16 Apr',
       '26 Jun', '29 May', '29 Jun', '29 Mar', '23 May', '17 Jun'],
      dtype=object)

In [67]:
pd.to_datetime(flights.Arrival_Time).dt.time

C:\Users\Dell\AppData\Local\Temp\ipykernel_1260\2994693450.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  pd.to_datetime(flights.Arrival_Time).dt.time


0        01:10:00
1        13:15:00
2        04:25:00
3        23:30:00
4        21:35:00
           ...   
10678    22:25:00
10679    23:20:00
10680    11:20:00
10681    14:10:00
10682    19:15:00
Name: Arrival_Time, Length: 10682, dtype: object

## Duration 

In [68]:
flights.Duration

0        2h 50m
1        7h 25m
2           19h
3        5h 25m
4        4h 45m
          ...  
10678    2h 30m
10679    2h 35m
10680        3h
10681    2h 40m
10682    8h 20m
Name: Duration, Length: 10682, dtype: object

In [69]:
(
    flights
    .Duration
    .loc[lambda ser: ~ser.str.contains("m")]
    #              ☝️
    #     this symbol separates all the rows which does not contain "m"
    .unique()
)

array(['19h', '23h', '22h', '12h', '3h', '5h', '10h', '18h', '24h', '15h',
       '16h', '8h', '14h', '20h', '13h', '11h', '9h', '27h', '26h', '4h',
       '7h', '30h', '21h', '28h', '47h', '6h', '25h', '38h', '34h'],
      dtype=object)

In [70]:
(
    flights
    .Duration
    .loc[lambda ser: ~ser.str.contains("h")]
    #              ☝️
    #     this symbol separates all the rows which does not contain "m"
    #.unique()
)

6474    5m
Name: Duration, dtype: object

In [71]:
flights.iloc[[6474]]

Airline Date_of_Journey    Source Destination      Route Dep_Time  \
6475  Air India      09/03/2019  Banglore   New Delhi  BLR → DEL    17:00   

     Arrival_Time Duration Total_Stops Additional_Info  Price  
6475        19:50   2h 50m    non-stop         No info   9481

- The observation indexed 6474 has duration of 5 minutes. This is clearly wrong. Will delete this observation.

In [72]:
(
    flights
    .Duration
    .drop(index=6474)
    .str.split(" ",expand=True)
    .set_axis(["hour","minute"], axis=1)
    .assign(
        hour=lambda df_: (
            df_
            .hour
            .str.replace("h", "")
            .astype(int)
            .mul(60)
        ),
        minute= lambda df_: (
            df_
            .minute
            .str.replace("m","")
            .fillna("0")
            .astype(int)
        )
    )
    .sum(axis=1)
    .rename("duration_minutes")
    .to_frame()
    #.dtypes
    #.isna().sum()
)

duration_minutes
0                   170
1                   445
2                  1140
3                   325
4                   285
...                 ...
10678               150
10679               155
10680               180
10681               160
10682               500

[10681 rows x 1 columns]

## Total Stops

In [73]:
flights.Total_Stops.unique()

array(['non-stop', '2 stops', '1 stop', '3 stops', nan, '4 stops'],
      dtype=object)

In [74]:
(
    flights
    .Total_Stops
    .replace("non-stop", "0")
    .str.replace(" stops?", "", regex=True)# this will search for both either 'stops' or stop
    #.unique()
    .pipe(lambda ser: pd.to_numeric(ser))
)

0        0.0
1        2.0
2        2.0
3        1.0
4        1.0
        ... 
10678    0.0
10679    0.0
10680    0.0
10681    0.0
10682    2.0
Name: Total_Stops, Length: 10682, dtype: float64

In [75]:
flights.Additional_Info.unique()

array(['No info', 'In-flight meal not included',
       'No check-in baggage included', '1 Short layover', 'No Info',
       '1 Long layover', 'Change airports', 'Business class',
       'Red-eye flight', '2 Long layover'], dtype=object)

## 5. Cleaning Data

In [76]:
def convert_to_minutes(ser):
    return (
        ser
        .str.split(" ",expand=True)
        .set_axis(["hour","minute"], axis=1)
        .assign(
            hour=lambda df_: (
            df_
            .hour
            .str.replace("h", "")
            .astype(int)
            .mul(60)
        ),
        minute= lambda df_: (
            df_
            .minute
            .str.replace("m","")
            .fillna("0")
            .astype(int)
        )
    )
    .sum(axis=1)
    .rename("duration_minutes")
    .to_frame()
        
    )
    

**Dictionary unpacking** in Python is a powerful feature that allows you to extract key-value pairs from a dictionary and assign them to variables in a single step. The most common method for unpacking dictionaries is by using the ** (double asterisk) operator, which can be used in various scenarios, such as:

In [77]:
def clean_data(df):
    return (
        df
        .drop(index = [6474])
        .drop_duplicates()
        .assign(**{
            col: df[col].str.strip() 
            # ☝️ The strip() function in pandas is used to remove leading and trailing whitespace characters from strings in a 
            # Series or DataFrame column. It is typically applied using .str.strip() for string operations.
            for col in df.select_dtypes(include="O").columns 
            # ☝️ here we are selecting columns which are of object('O') type 
        })
        .rename(columns=str.lower)
        .assign(
            airline=lambda df_: (
                df_
                .airline
                .str.replace(" Premium economy", "")
                # ☝️ here we replaced the string 'Premium economy' with empty string
                .str.replace(" Business", "")
                # ☝️ here we replaced the string 'Business' with empty string
                .str.title()
                # ☝️ the first letter of the titles will be made capital . i.e carriers->Carriers
            ),
            date_of_journey=lambda df_: pd.to_datetime(df_.date_of_journey, dayfirst=True),
            dep_time=lambda df_: pd.to_datetime(df_.dep_time).dt.time,
            arrival_time=lambda df_: pd.to_datetime(df_.arrival_time).dt.time,
            duration = lambda df_: df_.duration.pipe(convert_to_minutes),
            total_stops = lambda df_: (
                df_
                .total_stops
                .replace("non-stop", "0")
                .str.replace(" stops?", "", regex=True)# this will search for both either 'stops' or stop
                #.unique()
                .pipe(lambda ser: pd.to_numeric(ser))
            ),
            additional_info=lambda df_: df_.additional_info.replace("No info", "No Info")
        )
        .drop(columns="route")
    )

In [78]:
flights_cleaned = clean_data(flights)#.sample(5)
flights_cleaned

C:\Users\Dell\AppData\Local\Temp\ipykernel_1260\1191007153.py:26: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  dep_time=lambda df_: pd.to_datetime(df_.dep_time).dt.time,
C:\Users\Dell\AppData\Local\Temp\ipykernel_1260\1191007153.py:27: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  arrival_time=lambda df_: pd.to_datetime(df_.arrival_time).dt.time,


airline date_of_journey    source destination  dep_time  \
0           Indigo      2019-03-24  Banglore   New Delhi  22:20:00   
1        Air India      2019-05-01   Kolkata    Banglore  05:50:00   
2      Jet Airways      2019-06-09     Delhi      Cochin  09:25:00   
3           Indigo      2019-05-12   Kolkata    Banglore  18:05:00   
4           Indigo      2019-03-01  Banglore   New Delhi  16:50:00   
...            ...             ...       ...         ...       ...   
10678     Air Asia      2019-04-09   Kolkata    Banglore  19:55:00   
10679    Air India      2019-04-27   Kolkata    Banglore  20:45:00   
10680  Jet Airways      2019-04-27  Banglore       Delhi  08:20:00   
10681      Vistara      2019-03-01  Banglore   New Delhi  11:30:00   
10682    Air India      2019-05-09     Delhi      Cochin  10:55:00   

      arrival_time  duration  total_stops additional_info  price  
0         01:10:00       170          0.0         No Info   3897  
1         13:15:00       445          2.0         No Info   7662  
2         04:25:00      1140          2.0         No Info  13882  
3         23:30:00       325          1.0         No Info   6218  
4         21:35:00       285          1.0         No Info  13302  
...            ...       ...          ...             ...    ...  
10678     22:25:00       150          0.0         No Info   4107  
10679     23:20:00       155          0.0         No Info   4145  
10680     11:20:00       180          0.0         No Info   7229  
10681     14:10:00       160          0.0         No Info  12648  
10682     19:15:00       500          2.0         No Info  11753  

[10461 rows x 10 columns]

In [79]:
clean_data(flights).airline.unique()

C:\Users\Dell\AppData\Local\Temp\ipykernel_1260\1191007153.py:26: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  dep_time=lambda df_: pd.to_datetime(df_.dep_time).dt.time,
C:\Users\Dell\AppData\Local\Temp\ipykernel_1260\1191007153.py:27: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  arrival_time=lambda df_: pd.to_datetime(df_.arrival_time).dt.time,


array(['Indigo', 'Air India', 'Jet Airways', 'Spicejet',
       'Multiple Carriers', 'Goair', 'Vistara', 'Air Asia', 'Trujet'],
      dtype=object)

# 6. Split the data 

In [80]:
flights_final = flights_cleaned.sample(1000)

In [81]:
X = flights_final.drop(columns=['price'])
y = flights_final['price']

When building ML models, it's important to split your data into separate sets:

- Training set → to train the model.
- Validation set → to tune the model and check its performance during training.
- Test set → to evaluate the final model’s performance on unseen data.

In [82]:
X_, X_test, y_, y_test = train_test_split(X,y, test_size=0.2, random_state=42)

This splits your entire dataset into:

- 80% → X_, y_ (used for training + validation)
- 20% → X_test, y_test (used only for final testing)
- random_state=42 ensures the split is reproducible (you'll get the same result every time you run the code).
- X = input features
- y = labels/targets



In [83]:
X_train, X_val, y_train, y_val = train_test_split(X_,y_, test_size=0.2, random_state=42)

Now, from the 80% you kept (X_, y_), you're splitting it further:

- 80% of 80% = 64% → X_train, y_train (used to train the model)

- 20% of 80% = 16% → X_val, y_val (used to validate the model while training)

In [84]:
print(X_train.shape, y_train.shape)
print(X_val.shape, y_val.shape)
print(X_test.shape, y_test.shape)

(640, 9) (640,)
(160, 9) (160,)
(200, 9) (200,)


# 7. Export the subsets

In [85]:
def export_data(X,y,name):
    file_name = f"{name}.csv"
    file_path = os.path.join(PROJECT_DIR,DATA_DIR, file_name)

    X.join(y).to_csv(file_path, index = False)

    return pd.read_csv(file_path).head()

In [86]:
export_data(X_train,y_train,"train")

airline date_of_journey    source destination  dep_time  \
0          Air India      2019-03-06    Mumbai   Hyderabad  06:00:00   
1  Multiple Carriers      2019-05-18     Delhi      Cochin  13:00:00   
2        Jet Airways      2019-03-21  Banglore   New Delhi  11:40:00   
3          Air India      2019-05-18   Kolkata    Banglore  12:00:00   
4              Goair      2019-05-03  Banglore       Delhi  07:45:00   

  arrival_time  duration  total_stops              additional_info  price  
0     09:10:00      1630          2.0                      No Info  13169  
1     01:30:00       750          1.0                      No Info   9385  
2     10:25:00      1365          1.0  In-flight meal not included   7832  
3     18:30:00      1830          2.0                      No Info  10361  
4     10:40:00       175          0.0                      No Info   3898

In [87]:
export_data(X_val, y_val,"valid")

airline date_of_journey    source destination  dep_time arrival_time  \
0  Jet Airways      2019-03-12    Mumbai   Hyderabad  03:00:00     16:35:00   
1     Spicejet      2019-05-24  Banglore       Delhi  05:55:00     08:35:00   
2    Air India      2019-05-15   Kolkata    Banglore  09:10:00     04:40:00   
3  Jet Airways      2019-05-27     Delhi      Cochin  14:00:00     04:25:00   
4  Jet Airways      2019-06-06     Delhi      Cochin  09:40:00     12:35:00   

   duration  total_stops              additional_info  price  
0       815          1.0                      No Info  22294  
1       160          0.0                      No Info   3311  
2      1170          2.0                      No Info   6117  
3       865          1.0  In-flight meal not included  12898  
4      1615          2.0  In-flight meal not included  11733

In [88]:
export_data(X_test,y_test,"test")

airline date_of_journey   source destination  dep_time arrival_time  \
0  Jet Airways      2019-06-24   Mumbai   Hyderabad  02:55:00     04:20:00   
1  Jet Airways      2019-03-03    Delhi      Cochin  21:50:00     18:50:00   
2  Jet Airways      2019-05-06  Kolkata    Banglore  17:00:00     09:45:00   
3       Indigo      2019-03-21  Kolkata    Banglore  15:10:00     17:45:00   
4  Jet Airways      2019-06-03    Delhi      Cochin  09:35:00     12:35:00   

   duration  total_stops              additional_info  price  
0        85          0.0  In-flight meal not included   4050  
1      1260          1.0                      No Info  17024  
2      1005          1.0  In-flight meal not included   8066  
3       155          0.0                      No Info   4148  
4      1620          2.0                      No Info  13292